src/chat/chat_panel.rs

```diff
@@ -468,122 +471,43 @@ pub struct ChatPanel {
     view: View,
 
     #[rust]
-    state: ChatPanelState,
+    state: State,
 }
 
 impl Widget for ChatPanel {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
         self.view.handle_event(cx, event, scope);
         self.widget_match_event(cx, event, scope);
+        self.update_state(scope);
+        dbg!(self.state);
```

Comment:
I will leave this line here until someone else validates the new changes as it gives visibility of the state.